### PR TITLE
Multinode objectstorage

### PR DESCRIPTION
**Pull request by mephenor**
This is the final of the multinode storage config PRs.

As we have not yet decided on a mechanism to select which storage to use and which service needs to be responsible, I need some feedback on the endpoint change for `POST  uploads`.
For now this one is changed to expect the endpoint_alias for the storage node where the uploaded data should reside.

It might be reasonable to additionally keep the older version around, using some pre-configured storage alias (i.e. not hard coded), but this masks the storage location and probably adds a layer of confusion about which one should actually be used. With the other endpoint exposed now, I'm not sure how RESTful this approach would be, as both endpoints basically do the same.

Another way to deal with the issue for now might be reverting that endpoint change and just going with the pre-configured alias for now.

### CODE REVIEW

File: openapi.yaml [hunk 1]
```diff
@@ -434,10 +464,17 @@ paths:
       summary: health
       tags:
       - UploadControllerService
-  /uploads:
+  /uploads/storages/{storage_alias}:
```
Comment (KerstenBreuer) [inline openapi.yaml]: The storage alias should rather go into the body.

File: openapi.yaml [hunk 2]
```diff
@@ -477,6 +514,15 @@ paths:
 
             - noFileAccess: The user is not registered as a Data Submitter for the
             corresponding file.'
+        '404':
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/HttpUnknownStorageAliasError'
```
Comment (KerstenBreuer) [inline openapi.yaml:520]: Because of the above, the code for this error should be 400.

File: src/ucs/core/models.py [hunk 3]
```diff
@@ -69,6 +69,10 @@ class UploadAttempt(BaseModel):
         ..., description="The public key used by the submittter to encrypt the file."
     )
     model_config = ConfigDict(from_attributes=True, title="Multi-Part Upload Details")
+    s3_endpoint_alias: str = Field(
```
Comment (KerstenBreuer) [inline src/ucs/core/models.py]: Just by convention the core should not name the technology:
```suggestion
    storage_alias: str = Field(
```

**Review by KerstenBreuer — CHANGES_REQUESTED** [review]
Looks good, I think the solution specifying the alias in the post request is good.

File: openapi.yaml (hunk 1 above)
Comment (mephenor) [reply openapi.yaml]: I don't know why, but I forgot about that possibility, as that directly resolves a large part of my problem...
Thanks and changed

File: src/ucs/core/models.py (hunk 3 above)
Comment (mephenor) [reply src/ucs/core/models.py]: Probably needs some changes across other services as well.
Now only the outbound event payload knows that it's relating to S3 and it's storage_alias everywhere else

File: openapi.yaml (hunk 2 above)
Comment (mephenor) [reply openapi.yaml:520]: Probably also needs an additional test, I'm on it

**Review by KerstenBreuer — APPROVED** [review]

**Pull request merged**